### PR TITLE
Security - Backend - wolfi container base image with support for rootless user

### PR DESCRIPTION
**Pull request by yashgorana**
## Description

Part of https://github.com/OpenMined/Heartbeat/issues/822

* Update backend to work with `nonroot` linux user
* Faster build times by caching ML/Pytorch dependencies

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/setup.cfg [hunk 1]
```diff
@@ -56,15 +49,24 @@ syft =
...
+    jax==0.4.18
+    opendp==0.8.0
     safetensors==0.4.0
     transformers==4.34.0
     evaluate==0.4.1
+    dm-haiku==0.0.10
     torch==2.1.0
     recordlinkage==0.16
-    argon2-cffi==23.1.0
+    # =====  DOCKER:CACHED:END ===== #
```
Comment (yashgorana) [inline packages/syft/setup.cfg]: I'm clubbing these big packages between these `DOCKER:CACHED` hints.

`backend.dockerfile` will pick these up, install them and cache them, and any changes to `setup.cfg` will not trigger a re-download unless these are updated explicitly. The remaining packages will be installed in the final `pip install -e syft` stage

File: packages/grid/backend/grid/core/node.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 mongo_store_config = MongoStoreConfig(client_config=mongo_client_config)
 
 
-client_config = SQLiteStoreClientConfig(path="/storage/")
```
Comment (yashgorana) [inline packages/grid/backend/grid/core/node.py:29]: @madhavajay is this supposed to be `BACKEND_STORAGE_PATH` env? If yes then I need to make the fixes across the compose files as well.

File: packages/grid/worker/worker.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 worker = worker_class(
     name=node_name,
     local_db=True,
-    sqlite_path="/storage/",
```
Comment (yashgorana) [inline packages/grid/worker/worker.py:37]: Same over here - `BACKEND_STORAGE_PATH`?

File: packages/grid/backend/grid/core/node.py (hunk 2 above)
Comment (yashgorana) [reply packages/grid/backend/grid/core/node.py:29]: I figured that `BACKEND_STORAGE_PATH` is not being used in syft. It's just used in hagrid to tweak the host volume path for credentials. I'll tweak the name of this to be more indicative of this.

Comment (yashgorana) [reply packages/grid/backend/grid/core/node.py:29]: Additionally, these `/storage` also need to be made rootless. I'll make them `$HOME/data`

**Review by madhavajay — APPROVED** [review]
So fresh and squeaky clean! Amazing work @yashgorana.
I tested it on arm64 and it works great. I think we can use this for all our containers going forward. ❤️🙏

**Comment by yashgorana** [comment]
I missed out on posting the security report:
Screenshot 2023-10-26 233705

**Pull request merged**